backend/abstract/processor.py

```diff
@@ -71,19 +71,20 @@ def is_compatible_with(cls, module=None):
 	#: Extension of the file created by the processor
 	extension = "csv"
 
-	#: Configurable options for this processor
-	options = {}
-
-	#: Values for the processor's options, populated by user input
-	parameters = {}
-
 	#: Is this processor running 'within' a preset processor?
 	is_running_in_preset = False
 
 	#: This will be defined automatically upon loading the processor. There is
 	#: no need to override manually
 	filepath = None
 
+	# def __init__(self, logger, job, queue=None, manager=None, modules=None):
```

Comment:
I left this year for the moment as a comment. It does not _seem_ necessary for the base processor class to have either `options` or `parameters` as they are updated when needed. 

`options` in particular is always retrieved by the `get_options` method which correctly returns a blank dictionary if the child class does not define `options` itself.

Reply:
That makes sense, most of the set-up for `BasicProcessor` happens in `work()` anyway since it is always instantiated in a separate thread. So insofar as these properties need to be set up that is where it would happen and it is also where `self.parameters` is set.